Lib/test/_test_multiprocessing.py

```diff
@@ -3995,9 +3995,17 @@ def test_shared_memory_ShareableList_basics(self):
         sl[4] = 'some'  # Change type at a given position.
         self.assertEqual(sl[4], 'some')
         self.assertEqual(sl.format, '8s8sdq8sxxxxxxx?q')
-        with self.assertRaises(ValueError):
-            sl[4] = 'far too many'  # Exceeds available storage.
+        with self.assertRaisesRegex(ValueError,
+                                    "exceeds available storage"):
+            sl[4] = 'far too many'
         self.assertEqual(sl[4], 'some')
+        sl[0] = 'encodés'  # Exactly 8 bytes of UTF-8 data
+        self.assertEqual(sl[0], 'encodés')
+        self.assertEqual(sl[1], b'HoWdY')  # no spillage
+        with self.assertRaisesRegex(ValueError,
+                                    "exceeds available storage"):
+            sl[0] = 'encodées'  # Exactly 9 bytes of UTF-8 data
```

Comment:
Could we also add a test for exceeding the storage with a byte string? E.g.:

```suggestion
        with self.assertRaisesRegex(ValueError,
                                    "exceeds available storage"):
            sl[0] = 'encodées'  # Exactly 9 bytes of UTF-8 data
        with self.assertRaisesRegex(ValueError,
                                    "exceeds available storage"):
            sl[1] = b'HoWdDyYyY'  # Exactly 9 bytes
```

It's more of a sanity test than anything, but I think it makes sense to include that as well.

Reply:
Will do, thank you.